### ISSUE TITLE
Fetching required data before transitioning

### ISSUE THREAD

**Issue by martindederer** [zxbodya/router1#1]
You mentioned in the readme file that is possible to fetch/wait for required data before transitioning. Do you have a concise example for that? I trying to wrap my head around how to do that with router1 unsuccessfully for a while now.

**Comment by zxbodya** [comment]
Not yet... But, I am planning to add better examples for core features like this one(hope to do it somewhere soon)

But anyway, before I have it:

there is `createHandler` method in router configuration, as I mentioned in readme it:

> should return an Observable which emits value when state is ready to be rendered, 
>  or when handler does not support that request:
>  - false: if handler can not redner this state - for example data not found 
>      or next matcher route handler should be used
>  - state handler, when data was loaded and can be rendered, object with methods

That means observable you create there can fetch data and emit state handler or false basing on this data, and transition will happen only after this observable will emit a value.

So schematically - it would be something like:

```js
function createHandler(transition){
  fetchTransitionData(transition).map(data => {
    if (!data) return false;
    return createStateHandlerWithData(data);
  })
}
```

It is up to you how to implement `fetchTransitionData` - here you have all the information resolved by router.

**Comment by martindederer** [comment]
I see. So if i would want to fetch/wait/do something different depending on the  supplied transition data structure i would have to perform this dispatch in createHandler? I was kind of hoping that this would be a configurable per route.

**Comment by zxbodya** [comment]
No, this is entirely possible to do on per route basis(and generally it is intended to do it that way), for example if for route handler you return an observable that waits for data before emitting something renderable. Than you can just return that in createHandler…

But with create handler - you can combine nested routes in a way you want in and have some extra possibilities.

---

For example in https://github.com/zxbodya/router1-app-template, you can modify this handler: https://github.com/zxbodya/router1-app-template/blob/master/src/homePage/homeHandler.js

as following:

```js
export default function () {
  return fetchData().map(data => ({
    meta: {
      title: 'Home page',
      description: 'home page description',
    },
    view: createContainer(Home, data),
  }));
}
```

And this way router will wait for `fetchData()` to complete before navigating to that page

**Comment by martindederer** [comment]
Ok, i think i got your point. I'll have to try that.

Thanks. :)
